serde_codegen/src/ser.rs

```diff
@@ -253,8 +255,12 @@ fn serialize_struct(
     );
 
     let type_name = name_expr(builder, item_attrs.name());
-    let len = fields.iter()
+
+    let serialized_fields: Vec<_> = fields.iter()
```

Comment:
we could eliminate the allocation by folding into a tuple, not sure if that makes any difference in readability or performance, so feel free to merge as is.